app/screens/channel/channel_base.js

```diff
@@ -255,14 +256,47 @@ export default class ChannelBase extends PureComponent {
             theme,
         } = this.props;
 
+        const style = getStyleFromTheme(theme);
+
         if (!currentChannelId) {
             if (channelsRequestFailed) {
-                const PostListRetry = require('app/components/post_list_retry').default;
+                const FailedNetworkAction = require('app/components/failed_network_action').default;
+                const FormattedText = require('app/components/formatted_text').default;
+
+                const errorDescription = {
+                    id: t('mobile.failed_network_action.shortDescription'),
+                    defaultMessage: '{type} will load when you have an internet connection or {refresh}.',
+                    values: {
+                        type: (
+                            <FormattedText
+                                id='mobile.failed_network_action.description.messages'
+                                defaultMessage='messages'
+                            />
+                        ),
+                        refresh: (
+                            <FormattedText
+                                id='mobile.failed_network_action.retry'
+                                defaultMessage='try again'
+                                style={style.link}
+                                onPress={this.retryLoadChannels}
```

Comment:
if you end up passing this action as a prop of `<FailedNetworkAction>` then you don't need the changes to the styles here, then you could revert the changes in `channel.ios.js` and `channel.android.js`